File: examples-sdk/csharp/406-agent-vil-handler-shm/Main.cs
// 406-agent-vil-handler-shm — C# SDK equivalent
// Compile: vil compile --from csharp --input 406-agent-vil-handler-shm/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("fraud-detection-agent", 3126);
var fraud_agent = new ServiceProcess("fraud-agent");
fraud_agent.Endpoint("POST", "/detect", "detect_fraud");
fraud_agent.Endpoint("GET", "/health", "health");
server.Service(fraud_agent);
server.Compile();

File: examples-sdk/swift/406-agent-vil-handler-shm/main.swift
// 406-agent-vil-handler-shm — Swift SDK equivalent
// Compile: vil compile --from swift --input 406-agent-vil-handler-shm/main.swift --release

let server = VilServer(name: "fraud-detection-agent", port: 3126)
let fraud_agent = ServiceProcess(name: "fraud-agent")
fraud_agent.endpoint(method: "POST", path: "/detect", handler: "detect_fraud")
fraud_agent.endpoint(method: "GET", path: "/health", handler: "health")
server.service(fraud_agent)
server.compile()
